paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentOptionsUI.kt

```diff
@@ -0,0 +1,269 @@
+        ) {
+            items(state.items) { item ->
+                val isEnabled = !isProcessing && (!isEditing || item.isEnabledDuringEditing)
+                val isSelected = item == state.selectedItem && !isEditing
+
+                PaymentOption(
+                    item = item,
+                    nameProvider = nameProvider,
+                    width = width,
+                    isEditing = isEditing,
+                    isEnabled = isEnabled,
+                    isSelected = isSelected,
+                    onAddCardPressed = onAddCardPressed,
+                    onItemSelected = onItemSelected,
+                    onItemRemoved = onItemRemoved,
+                    modifier = Modifier.testTag(item.viewType.name)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+internal fun rememberItemWidth(maxWidth: Dp): Dp {
+    val targetWidth = maxWidth - 17.dp * 2
+    val minItemWidth = 100.dp + (6.dp * 2)
+    // numVisibleItems is incremented in steps of 0.5 items (1, 1.5, 2, 2.5, 3, ...)
+    val numVisibleItems = (targetWidth * 2 / minItemWidth).toInt() / 2f
+    return (targetWidth / numVisibleItems)
+}
+
+@Composable
+private fun PaymentOption(
+    item: PaymentOptionsItem,
+    nameProvider: (PaymentMethodCode?) -> String?,
+    width: Dp,
+    isEnabled: Boolean,
+    isEditing: Boolean,
+    isSelected: Boolean,
+    onAddCardPressed: () -> Unit,
+    onItemSelected: (PaymentSelection?) -> Unit,
+    onItemRemoved: (PaymentMethod) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    when (item) {
+        is PaymentOptionsItem.AddCard -> {
+            AddCard(
+                width = width,
+                isEnabled = isEnabled,
+                onAddCardPressed = onAddCardPressed,
+                modifier = modifier,
+            )
+        }
+        is PaymentOptionsItem.GooglePay -> {
+            GooglePay(
+                width = width,
+                isEnabled = isEnabled,
+                isSelected = isSelected,
+                onItemSelected = onItemSelected,
+                modifier = modifier,
+            )
+        }
+        is PaymentOptionsItem.Link -> {
+            Link(
+                width = width,
+                isEnabled = isEnabled,
+                isSelected = isSelected,
+                onItemSelected = onItemSelected,
+                modifier = modifier,
+            )
+        }
+        is PaymentOptionsItem.SavedPaymentMethod -> {
+            SavedPaymentMethod(
+                paymentMethod = item,
+                nameProvider = nameProvider,
+                width = width,
+                isEnabled = isEnabled,
+                isEditing = isEditing,
+                isSelected = isSelected,
+                onItemSelected = onItemSelected,
+                onItemRemoved = onItemRemoved,
+                modifier = modifier,
+            )
+        }
+    }
+}
+
+@Composable
+private fun AddCard(
```

Comment:
Should we add screenshot tests for the rest of these? I saw we're testing most of these (I think all of them but link) in aggregate. Is it worth adding screenshot tests for these in isolation?